src/main/java/seedu/healthmate/HistoryTracker.java

```
@@ -0,0 +1,196 @@
+        MealEntriesList mealEntriesList = new MealEntriesList();
+        for (Meal meal : meals) {
+            mealEntriesList.addMealWithoutCLIMessage(meal);
+        }
+        UI.printString("Meal Entries Loaded Successfully!");
+        return mealEntriesList;
+    }
+
+    public MealList loadMealOptions() {
+        List<Meal> meals = loadMealFromFile(MEAL_OPTIONS_FILE, false);
+        MealList mealList = new MealList();
+        for (Meal meal : meals) {
+            mealList.addMealWithoutCLIMessage(meal);
+        }
+        UI.printString("Meal Options Loaded Successfully!");
+        return mealList;
+    }
+
+    public Optional<User> loadUserData() {
+        try {
+            File userDataFile = new File(DATA_DIRECTORY + File.separator + USER_DATA_FILE);
+            if (!userDataFile.exists()) {
+                userDataFile.createNewFile();
+            }
+
+            Optional<User> user = loadUserFromFile(userDataFile);
+            return user;
+        } catch (IOException e) {
+            System.out.println("Error creating user data file: " + e.getMessage());
+        }
+        return Optional.empty();
+    }
+
+    public Optional<User> loadUserFromFile(File userDataFile) throws FileNotFoundException{
+        try {
+            Scanner s = new Scanner(userDataFile);
+            double userHeight = Double.parseDouble(s.nextLine());
+            double userWeight = Double.parseDouble(s.nextLine());
+            boolean isMale = Boolean.parseBoolean(s.nextLine());
+            int age = Integer.parseInt(s.nextLine());
+            String goal = s.nextLine();
+
+            return Optional.of(new User(userHeight, userWeight, isMale, age, goal));
+
+        } catch (NumberFormatException e) {
+            System.out.println("Error parsing a number." + e.getMessage());
+        } catch (NoSuchElementException e) {
+            // silent catch if existing user file contains no content
+        }
```

Try not to use empty catch block, you might want to implement some error catching/logging in the catch block.